OSDOCS-8694 z-stream RN for 4.14.2 (#67762)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

====

[id="ocp-4-14-2"]
=== RHSA-2023:6837 - {product-title} 4.14.2 bug fix and security update

Issued: 2023-11-15

{product-title} release 4.14.2, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2023:6837[RHSA-2023:6837] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2023:6840[RHSA-2023:6840] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.14.2 --pullspecs
----

[id="ocp-4-14-2-bug-fixes"]
==== Bug fixes

* Previously, a change to the default security settings for new Microsoft Azure storage accounts in the `eastus` region prevented the installation of {product-title} clusters that use Azure AD Workload Identity in that region. The issue has been resolved in this release. (link:https://issues.redhat.com/browse/OCPBUGS-22651[*OCPBUGS-22651*])

> * Previously, the Docker build deployment would fail since the inline Dockerfile hooks did not preserve the modification time of the file that was copied. With this release, the '-p' flag was added to the `cp` command when copying artifacts between containers to preserve timestamps. (link:https://issues.redhat.com/browse/OCPBUGS-23006[*OCPBUGS-23006*])

Reviewer: ```suggestion
* Previously, the Docker build deployment would fail because the inline Dockerfile hooks did not preserve the modification time of the file that was copied. With this release, the '-p' flag was added to the `cp` command when copying artifacts between containers to preserve timestamps. (link:https://issues.redhat.com/browse/OCPBUGS-23006[*OCPBUGS-23006*])
```

> * Previously, the Image Registry Operator made API calls to the Storage Account List endpoint as part of obtaining access keys every 5 minutes. In projects with many {product-title} (OCP) clusters this could lead to API limits being reached causing 429 errors when attempting to create new clusters. With this release, the time between calls is increased from 5 minutes to 20 minutes. (link:https://issues.redhat.com/browse/OCPBUGS-22127[*OCPBUGS-22127*])

Reviewer: ```suggestion
* Previously, the Image Registry Operator made API calls to the Storage Account List endpoint as part of obtaining access keys every 5 minutes. In projects with many {product-title} (OCP) clusters, this could lead to API limits being reached, causing 429 errors when attempting to create new clusters. With this release, the time between calls is increased from 5 minutes to 20 minutes. (link:https://issues.redhat.com/browse/OCPBUGS-22127[*OCPBUGS-22127*])
```
(Unsure about 2nd comma)

> * Previously, an Azure Managed Identity role was omitted from cloud-controller-manager (CCM) service account which meant that CCM could not properly manage Service type LoadBalancers in environments deployed to existing vnets with private publishing method. With this release, the missing role was added to ccoctl so it is possible for Azure Managed Identity installations into existing vnet with private publishing. (link:https://issues.redhat.com/browse/OCPBUGS-21926[*OCPBUGS-21926*])

Reviewer: ```suggestion
* Previously, an Azure Managed Identity role was omitted from the cloud-controller-manager (CCM) service account, which meant that CCM could not properly manage `Service` type `LoadBalancers` in environments deployed to existing VNets with the private publishing method. With this release, the missing role was added to the CCO utility (`ccoctl`) so it is possible for Azure Managed Identity installations into existing VNet with private publishing. (link:https://issues.redhat.com/browse/OCPBUGS-21926[*OCPBUGS-21926*])
```
Not sure about `Service` or `LoadBalancers` markup but I found some in-repo examples that seemed to match. Or maybe it's "`Service` objects with type `LoadBalancer`" (found a different in-repo instance of that). 🤷 

Seems like "VNet" is the common spelling for general references tho.

* This patch enables egress IP for Azure setups that use outbound rules to achieve outbound connectivity. An architectural constraint in Azure prevents the secondary IP acting as egress IP from having outbound connectivity in such setups. This means that matching pods will have no outbound connectivity to the internet, but will be able to reach external servers in the infrastructure network, which is the intended use case for egress IP. (link:https://issues.redhat.com/browse/OCPBUGS-21785[*OCPBUGS-21785*])

> * Previously, when MetalLB Operator's controller restarts while having an IP assigned and unassigned LB services it restarts with an empty internal state which can break workloads. With this release, metallb's controller is modified to process the services with already have an assigned IP first. (link:https://issues.redhat.com/browse/OCPBUGS-16267[*OCPBUGS-16267*])

Reviewer: ```suggestion
* Previously, when MetalLB Operator's controller restarted while having an IP assigned and unassigned load balancer services, it restarted with an empty internal state, which can break workloads. With this release, MetalLB's controller is modified to first process the services that already have an assigned IP. (link:https://issues.redhat.com/browse/OCPBUGS-16267[*OCPBUGS-16267*])
```
